lib/std/io/buffered_reader.zig

```diff
@@ -39,6 +39,29 @@ pub fn BufferedReader(comptime buffer_size: usize, comptime ReaderType: type) ty
             return dest.len;
         }
 
+        /// Peek reads bytes into `dest` and returns the amount read without advancing the reader.
+        /// If the `dest` size is too small for the desired peek amount, partially fill `dest`
+        /// with as much data as fits.
```

Comment:
The desired peek amount is determined by `dest.len`, right? How could it then be possible for `dest` to be too small?